ksql-common/src/main/java/io/confluent/ksql/util/timestamp/TimestampExtractionPolicyFactory.java

```diff
@@ -34,19 +36,19 @@ private TimestampExtractionPolicyFactory() {
   public static TimestampExtractionPolicy create(
       final KsqlConfig ksqlConfig,
       final LogicalSchema schema,
-      final Optional<String> timestampColumnName,
+      final Optional<ColumnRef> timestampColumnName,
       final Optional<String> timestampFormat
   ) {
     if (!timestampColumnName.isPresent()) {
       return new MetadataTimestampExtractionPolicy(getDefaultTimestampExtractor(ksqlConfig));
     }
 
-    final String fieldName = timestampColumnName.get().toUpperCase();
+    final ColumnRef col = timestampColumnName.get();
 
-    final Column timestampColumn = schema.findValueColumn(fieldName)
+    final Column timestampColumn = schema.findValueColumn(col)
         .orElseThrow(() -> new KsqlException(
             "The TIMESTAMP column set in the WITH clause does not exist in the schema: '"
-                + fieldName + "'"));
+                + col.toString(FormatOptions.noEscape()) + "'"));
```

Comment:
there is one, but the parameterless one should backquote everything for safety. You should be very certain that you don't want escapes rather than the other way around